app/scripts/metamask-controller.js

```diff
@@ -571,7 +565,7 @@ export default class MetamaskController extends EventEmitter {
       onPreferencesStateChange: this.preferencesController.store.subscribe.bind(
         this.preferencesController.store,
       ),
-      onNetworkStateChange: networkControllerMessenger.subscribe.bind(
+      onNetworkDidChange: networkControllerMessenger.subscribe.bind(
         networkControllerMessenger,
         'NetworkController:stateChange',
```

Comment:
We should be subscribed to `networkDidChange` here

Reply:
@Gudahtt The code changes you commented on are now part of v11.7.3 https://github.com/MetaMask/metamask-extension/pull/22354

I wanted to clarify on the point you made here though. Are you saying these lines of code should be changed to:
```
onNetworkDidChange: networkControllerMessenger.subscribe.bind(
    networkControllerMessenger,
    'NetworkController:networkDidChange',
)
```

Reply:
Fixed here https://github.com/MetaMask/metamask-extension/pull/22372